internal/postgres/report.go

```diff
@@ -0,0 +1,496 @@
+	w.WriteString("  --------------------------------------\n")
+	fmt.Fprintf(w, "  %6s %6s %6s %6s  %s\n", "schema", "data", "skip", "error", "statement")
+	w.WriteString("  --------------------------------------\n")
+	for _, x := range l {
+		s := conv.stats.statement[x.statement]
+		fmt.Fprintf(w, "  %6d %6d %6d %6d  %s\n", s.schema, s.data, s.skip, s.error, x.statement)
+	}
+	w.WriteString("See github.com/lfittl/pg_query_go/nodes for definitions of statement types\n")
+	w.WriteString("(lfittl/pg_query_go is the library we use for parsing pg_dump output).\n")
+	w.WriteString("\n")
+}
+
+func writeUnexpectedConditions(conv *Conv, w *bufio.Writer) {
+	reparseInfo := func() {
+		if conv.stats.reparsed > 0 {
+			fmt.Fprintf(w, "Note: there were %d pg_dump reparse events while looking for statement boundaries.\n\n", conv.stats.reparsed)
+		}
+	}
+	prHeading("Unexpected Conditions", w)
+	if len(conv.stats.unexpected) == 0 {
+		w.WriteString("There were no unexpected conditions encountered during processing.\n\n")
+		reparseInfo()
+		return
+	}
+	w.WriteString("For debugging only. This section provides details of unexpected conditions\n")
+	w.WriteString("encountered as we processed the pg_dump data. In particular, the AST node\n")
+	w.WriteString("representation used by the lfittl/pg_query_go library used for parsing\n")
+	w.WriteString("pg_dump output is highly permissive: almost any construct can appear at\n")
+	w.WriteString("any node in the AST tree. The list details all unexpected nodes and\n")
+	w.WriteString("conditions.\n")
+	w.WriteString("  --------------------------------------\n")
+	fmt.Fprintf(w, "  %6s  %s\n", "count", "condition")
+	w.WriteString("  --------------------------------------\n")
+	for s, n := range conv.stats.unexpected {
+		fmt.Fprintf(w, "  %6d  %s\n", n, s)
+	}
+	w.WriteString("\n")
+	reparseInfo()
+}
+
+// justifyLines writes s out to w, adding newlines between words
+// to keep line length under 'limit'. Newlines are indented
+// 'indent' spaces.
+func justifyLines(w *bufio.Writer, s string, limit int, indent int) {
+	n := 0
+	startOfLine := true
+	words := strings.Split(s, " ") // This only handles spaces (newlines, tabs ignored).
+	for _, x := range words {
+		if n+len(x) > limit && !startOfLine {
+			w.WriteString("\n")
+			w.WriteString(strings.Repeat(" ", indent))
+			n = indent
+			startOfLine = true
+		}
+		if startOfLine {
+			w.WriteString(x)
+			n += len(x)
+		} else {
+			w.WriteString(" " + x)
+			n += len(x) + 1
+		}
+		startOfLine = false
+	}
+}
+
+// pct prints a percentage representation of (total-bad)/total
+func pct(total, bad int64) string {
+	if bad == 0 || total == 0 {
+		return "100"
+	}
+	pct := 100.0 * float64(total-bad) / float64(total)
+	if pct > 99.9 {
+		return fmt.Sprintf("%2.5f", pct)
+	}
+	if pct > 95.0 {
+		return fmt.Sprintf("%2.3f", pct)
+	}
+	return fmt.Sprintf("%2.0f", pct)
+}
+
+func prHeading(s string, w *bufio.Writer) {
```

Comment:
Rename to writeHeading and switch the argument order to match the standard for writers.

Reply:
Done.